hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metadata/MetadataTest.java

```diff
@@ -0,0 +1,449 @@
+        for (SingularAttribute<? super Person, ?> localId : ids) {
+            assertTrue(localId.isId());
+            assertSame(personType, localId.getDeclaringType());
+            assertSame(localId, personType.getDeclaredAttribute(localId.getName()));
+            assertSame(localId, personType.getDeclaredSingularAttribute(localId.getName()));
+            assertSame(localId, personType.getAttribute(localId.getName()));
+            assertSame(localId, personType.getSingularAttribute(localId.getName()));
+            assertTrue(personType.getAttributes().contains(localId));
+        }
+
+        final EntityType<Giant> giantType = scope.getEntityManagerFactory().getMetamodel().entity(Giant.class);
+        assertEquals("HomoGigantus", giantType.getName());
+        assertFalse(giantType.hasSingleIdAttribute());
+        final Set<SingularAttribute<? super Giant, ?>> giantIds = giantType.getIdClassAttributes();
+        assertNotNull(giantIds);
+        assertEquals(2, giantIds.size());
+        assertEquals(personType.getIdClassAttributes(), giantIds);
+        for (SingularAttribute<? super Giant, ?> localGiantId : giantIds) {
+            assertTrue(localGiantId.isId());
+            try {
+                giantType.getDeclaredAttribute(localGiantId.getName());
+                fail(localGiantId.getName() + " is a declared attribute, but shouldn't be");
+            } catch (IllegalArgumentException ex) {
+                // expected
+            }
+            try {
+                giantType.getDeclaredSingularAttribute(localGiantId.getName());
+                fail(localGiantId.getName() + " is a declared singular attribute, but shouldn't be");
+            } catch (IllegalArgumentException ex) {
+                // expected
+            }
+            assertSame(localGiantId, giantType.getAttribute(localGiantId.getName()));
+            assertTrue(giantType.getAttributes().contains(localGiantId));
+        }
+
+        final EntityType<FoodItem> foodType = scope.getEntityManagerFactory().getMetamodel().entity(FoodItem.class);
+        assertTrue(foodType.hasVersionAttribute());
+        final SingularAttribute<? super FoodItem, Long> version = foodType.getVersion(Long.class);
+        assertNotNull(version);
+        assertTrue(version.isVersion());
+        assertEquals(3, foodType.getDeclaredAttributes().size());
+
+    }
+
+    @Test
+    public void testBasic(EntityManagerFactoryScope scope) {
+        final EntityType<Fridge> entityType = scope.getEntityManagerFactory().getMetamodel().entity(Fridge.class);
+        final SingularAttribute<? super Fridge, Integer> singularAttribute = entityType.getDeclaredSingularAttribute(
+                "temperature",
+                Integer.class
+        );
+		assertEquals( Integer.class, singularAttribute.getBindableJavaType() );
+		assertEquals( Integer.class, singularAttribute.getType().getJavaType() );
+//        assertEquals(int.class, singularAttribute.getBindableJavaType());
```

Comment:
Can we remove these commented-out lines so we don't confused ourselves again? ;)

Reply:
I'm going to remove them, thanks @sebersole !